src/Models/Scopes/CrossselProductsScope.php

```diff
@@ -10,18 +10,15 @@ class CrossselProductsScope implements Scope
 {
     public function apply(Builder $builder, Model $model): void
     {
-        $builder->selectRaw('GROUP_CONCAT(DISTINCT related.entity_id) as amasty_crosssell_ids')
-            ->leftJoin('amasty_mostviewed_product_index as mainrule', function ($join) {
-                $join->on('mainrule.entity_id', '=', 'quote_item.product_id')
-                    ->where('mainrule.relation', 'where_show')
-                    ->where('mainrule.position', 'cart_into_crosssel')
-                    ->where('mainrule.store_id', config('rapidez.store'));
-            })
+        $builder->selectRaw('GROUP_CONCAT(DISTINCT related.entity_id) as amasty_related_ids')
             ->leftJoin('amasty_mostviewed_product_index as related', function ($join) {
-                $join->on('related.rule_id', '=', 'mainrule.rule_id')
+                $join->on('related.rule_id', '=', 'amasty_mostviewed_product_index.rule_id')
```

Comment:
If the intended functionality is to add these scopes to the model of this table we can now use
```suggestion
                $join->on('related.rule_id', '=', $model->qualifyColumn('rule_id'))
```
for all `amasty_mostviewed_product_index` columns